### PR TITLE
Migrate to Sound Null Safety

### PR DESCRIPTION
**Pull request by debkanchan**
@ammaratef45 please review the changes

### CODE REVIEW

File: lib/stack.dart [hunk 1]
```diff
@@ -24,27 +22,13 @@ class Stack<T> {
...
-  }
+  T top() => _list.last;
 
   /// get the size of the stack.
-  int size() {
-    return _list.length;
-  }
-
-  /// get the length of the stack.
-  int get length => size();
```
Comment (ammaratef45) [inline lib/stack.dart:37]: @DebkanchanSamadder Why did you delete `length` here?

**Review by ammaratef45 — CHANGES_REQUESTED** [review]
@DebkanchanSamadder Please see my comment

File: lib/stack.dart (hunk 1 above)
Comment (debkanchan) [reply lib/stack.dart:37]: As far as I know implicitly a stack has size not length. Moreover length and size do the same thing so it creates ambiguity.

**Comment by ammaratef45** [comment]
@DebkanchanSamadder looks like `test_coverage` (we depend on in our CI/CD), doesn't support null safety yet (https://github.com/pulyaevskiy/test-coverage/issues/33) 
I Will go ahead and merge #20 as it doesn't change any existing code

If you want we can still discuss the changes in the code you propose in this PR or you can open an issue

**Comment by debkanchan** [comment]
> @DebkanchanSamadder looks like `test_coverage` (we depend on in our CI/CD), doesn't support null safety yet (https://github.com/pulyaevskiy/test-coverage/issues/33) 
> I Will go ahead and merge #20 as it doesn't change any existing code
> 
> If you want we can still discuss the changes in the code you propose in this PR or you can open an issue

It's ok it's not a big deal you can close this PR if you want.

Please release an update as soon as you get a chance to.
